### PR TITLE
Added `macros.inheritsFrom`.

### LINKED ISSUE #6856: There is no easy way to check whether a NimNode Type inherits indirectly from another
This can be fixed by adding an `inheritsFrom(NimNode, NimNode): bool` in macros.nim. Can be used to fix #6805.

### PR DESCRIPTION
**Pull request by Veladus**
Fixed #6856.

### CODE REVIEW

File: lib/core/macros.nim [hunk 1]
```diff
@@ -448,6 +448,19 @@ proc expectLen*(n: NimNode, min, max: int) {.compileTime.} =
   if n.len < min or n.len > max:
     error("macro expects a node with " & $min & ".." & $max " children", n)
 
+proc inheritsFrom*(n:NimNode, base:NimNode) : bool {.compileTime.} =
```
Comment (Araq) [inline lib/core/macros.nim]: Format it like the other code in this file.

File: lib/core/macros.nim [hunk 2]
```diff
@@ -448,6 +448,19 @@ proc expectLen*(n: NimNode, min, max: int) {.compileTime.} =
...
+  ## Checks whether or not `n` has `base` as an direct or indirect super type.
+  expectKind(n, nnkObjectTy)
+  expectKind(base, nnkObjectTy)
+  var cur = n
+  while cur.kind != nnkEmpty:
+    cur = getType(cur[1])
+    if cur.sameType(base):
+      return true
+
+proc inheritsFrom*(n:NimNode, base:typedesc):bool {.compileTime.} =
```
Comment (Araq) [inline lib/core/macros.nim]: Same.

**Comment by Araq** [comment]
Add a test case.

**Comment by dom96** [comment]
ping @Veladus

**Comment by Araq** [comment]
No feedback and not important enough to spend more time on it. Closing.

**Pull request closed without merging**